packages/calcite-components/src/components/stepper/stepper.tsx

```diff
@@ -380,17 +470,86 @@ export class Stepper implements LocalizedComponent, T9nComponent {
     });
   }
 
-  private handleDefaultSlotChange = (event: Event): void => {
-    const slottedItems = slotChangeGetAssignedElements(event);
-    const items = slottedItems.filter((el) => el?.tagName === "CALCITE-STEPPER-ITEM");
-    const spacing = Array(items.length).fill("1fr").join(" ");
-    this.el.style.gridTemplateAreas = spacing;
-    this.el.style.gridTemplateColumns = spacing;
-    this.setStepperItemNumberingSystem();
+  private renderAction(position: Position): VNode {
+    const isPositionStart = position === "start";
+    const path = isPositionStart ? "chevron-left" : "chevron-right";
+    const { currentActivePosition, singleViewMode, layout } = this;
+    const dir = getElementDir(this.el);
+    const totalItems = this.items.length;
+    const id = `${this.guid}-${isPositionStart ? "start" : "end"}`;
+
+    return layout === "horizontal" && singleViewMode ? (
+      <calcite-action
+        alignment="center"
+        appearance="transparent"
+        class={{
+          [CSS.actionIcon]: true,
+        }}
+        compact={true}
+        data-position={position}
+        disabled={
+          (currentActivePosition === 0 && isPositionStart) ||
+          (currentActivePosition === totalItems - 1 && !isPositionStart)
+        }
+        icon={path}
+        iconFlipRtl={dir === "rtl"}
```

Comment:
this should be set to true and not depend on the dir. It handles the dir internally.